tests/artifact_key_filter_test.py

```diff
@@ -0,0 +1,92 @@
+import pyshelf.artifact_key_filter as akfilter
+from tests.unit_test_base import UnitTestBase
+
+
+class ArtifactKeyFilterTest(UnitTestBase):
```

Comment:
I think functionally is ok for that test case.  I am already using a fake key.  As long as we know the interface is the same, there is not a whole lot else I could change in the test to make it different.

However, testing functionally that those have the same interface would be good